src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,176 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> previous;
+
+        private Node(Node<T> previous, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.previous = previous;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong Index.");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index != 0 && index != size) {
```

Comment:
do we need to check `index != size` here? (look at the line 41)

Reply:
okey